packages/nimble-components/src/table/specs/table-action-menu-selection-hld.md

```diff
@@ -0,0 +1,234 @@
+# Table Selection Mode for Action Menus HLD
+
+## Problem Statement
+
+Currently, the Nimble table component automatically changes row selection when users open action menus on rows. This behavior creates usability issues in scenarios where users want to access row-specific actions without changing which rows are currently selected.
+
+This design proposes adding a configurable `action-menu-changes-selection` attribute to the table component that allows clients to control whether opening action menus automatically selects rows.
+
+## Links To Relevant Work Items and Reference Material
+
+[Configurable table selection mode for action menus](https://github.com/ni/nimble/issues/2708)
+
+## Implementation / Design
+
+### Overview
+
+This design adds a new boolean attribute `action-menu-changes-selection` to the `nimble-table` component with the following characteristics:
+
+- **Default value**: `true` (action menus change selection - preserves existing behavior)
+- **Backward compatibility**: Existing tables continue to work unchanged
+- **Opt-in new behavior**: Setting to `false` enables non-intrusive action menu behavior
+- **Enhanced event context**: New `operatingRecord` field provides trigger record information in both modes
```

Comment:
Can you clarify the name of the event this is referring to?

Reply:
I took this section out, and specifically list the new field under `TableActionMenuToggleEventDetail` in the API changes section